ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7990,19 +7998,29 @@ protected boolean enableColumnStatsCollecting() {
     return true;
   }
 
-  private Path getCtasLocation(CreateTableDesc tblDesc, boolean createTableWithSuffix) throws SemanticException {
+  private Path getCtasOrCMVLocation(CreateTableDesc tblDesc, CreateMaterializedViewDesc viewDesc,
+                               boolean createTableWithSuffix) throws SemanticException {
     Path location;
+    String protoName;
+    String[] names;
+    Table tbl;
     try {
-      String protoName = tblDesc.getDbTableName();
-      String[] names = Utilities.getDbTableName(protoName);
-
-      // Handle table translation initially and if not present
-      // use default table path.
-      // Property modifications of the table is handled later.
-      // We are interested in the location if it has changed
-      // due to table translation.
-      Table tbl = tblDesc.toTable(conf);
-      tbl = db.getTranslateTableDryrun(tbl.getTTable());
+      if (tblDesc != null) {
+        protoName = tblDesc.getDbTableName();
+        names = Utilities.getDbTableName(protoName);
```

Comment:
this part is the same for both

Reply:
can't we extract `names = Utilities.getDbTableName(protoName);`?

Reply:
Replaced with 
`names = Utilities.getDbTableName(tblDesc.getDbTableName());` - For CTAS
AND 
`names = Utilities.getDbTableName(viewDesc.getViewName());` - For CMV

Reply:
minor, however, could you please check how it's done in `getDefaultCtasOrCMVLocation`.  
Not sure, but don't those 2 have some common parts, if yes maybe we could refactor and move the method into Util class?

Reply:
@deniskuzZ I have refactored the common parts of these 2 functions to a new function in `Utilities.java`. Please review.